pkg/daemon/osrelease/osrelease.go

```diff
@@ -25,17 +37,79 @@ const (
 // OperatingSystem is a wrapper around a subset of the os-release fields
 // and also tracks whether ostree is in use.
 type OperatingSystem struct {
-	// id is the ID field from the os-release
+	// id is the ID field from the os-release or inferred from the OS image
+	// label.
 	id string
-	// variantID is the VARIANT_ID field from the os-release
+	// variantID is the VARIANT_ID field from the os-release or inferred from the
+	// OS image label.
 	variantID string
-	// version is the VERSION, RHEL_VERSION, or VERSION_ID field from the os-release
+	// version is the VERSION, RHEL_VERSION, or VERSION_ID field from the
+	// os-release or image label.
 	version string
-	// osrelease is the underlying struct from github.com/ashcrow/osrelease
-	osrelease osrelease.OSRelease
+	// values is a map of all the values we uncovered either via the
+	// /etc/os-release / /usr/lib/os-release files *or* the labels attached
+	// to an OS image.
+	values map[string]string
+	// source identifies whether this came from the OSRelease file or from image labels.
+	source InfoSource
+}
+
+func newOperatingSystemFromImageLabels(imageLabels map[string]string) (OperatingSystem, error) {
+	os := OperatingSystem{
+		values: imageLabels,
+		source: ImageLabelInfoSource,
+	}
+
+	if isCoreOSVariant(imageLabels) {
+		os.variantID = coreos
+	} else {
+		return os, fmt.Errorf("unable to identify coreos variant from labels: %v", imageLabels)
+	}
+
+	version := ""
+	versionOK := false
+	if version, versionOK = imageLabels["version"]; !versionOK {
```

Comment:
Let's also try to handle the standard OCI label for this.  See https://github.com/ostreedev/ostree-rs-ext/pull/454

Basically the standard key is `org.opencontainers.image.version` so let's look for that first.

Reply:
I don't see that key in any of the labels I used to set up the tests. Did I not get the right labels?